code/controllers/subsystem/air.dm

```diff
@@ -577,16 +577,13 @@ GLOBAL_LIST_EMPTY(colored_images)
 		CHECK_TICK
 
 
-/datum/controller/subsystem/air/proc/get_init_dirs(type, dir)
-	if(!pipe_init_dirs_cache[type])
-		pipe_init_dirs_cache[type] = list()
+/datum/controller/subsystem/air/proc/get_init_dirs(type, dir, init_dir)
 
-	if(!pipe_init_dirs_cache[type]["[dir]"])
-		var/obj/machinery/atmospherics/temp = new type(null, FALSE, dir)
-		pipe_init_dirs_cache[type]["[dir]"] = temp.GetInitDirections()
-		qdel(temp)
+	var/obj/machinery/atmospherics/temp = new type(null, FALSE, dir, init_dir)
+	var/got_dir = temp.GetInitDirections()
+	qdel(temp)
```

Comment:
not being able to cache this is hhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh
why can't we?

Reply:
This was cached for normal pipes, but smart pipes can't really cache this because the way they work, the same type of pipe can have many different attachments, and if I cache a pipe to have a specific init direction, when I try that same pipe type but with a different init direction it won't matter since its already cached and it won't behave as you think.

I don't really think caching this is necessary since is only called when you try to place a pipe and there aren't many pipe placing going in the round at the same time.

Reply:
can't you use init_dir as a part of the cache? or is it more of a structural problem

Reply:
ye, recached it